rdkit/Chem/Draw/cairoCanvas.py

```diff
@@ -10,44 +10,120 @@
 #  of the RDKit source tree.
 #
 #pylint: disable=F0401,C0324,C0322,W0142
-import sys
+import sys, os
 
 from rdkit import six
 if not six.PY3:
   bytes = buffer
 
+have_cairocffi = False
 # for Python3, import cairocffi preferably
 if six.PY3:
   try:
     import cairocffi as cairo
   except ImportError:
     import cairo
+  else:
+    have_cairocffi = True
 else:
   try:
     import cairo
   except ImportError:
-    import cairocffi as cairo
+    try:
+      import cairocffi as cairo
+    except:
+      raise
+    else:
+      have_cairocffi = True
+have_pango = False
+if not 'RDK_NOPANGO' in os.environ:
+  if (have_cairocffi):
+    import cffi
+    import platform
+    ffi = cffi.FFI()
+    ffi.cdef('''
+        /* GLib */
+        typedef void* gpointer;
+        typedef void cairo_t;
+        typedef void PangoFontDescription;
+        void g_object_unref (gpointer object);
 
+        /* Pango and PangoCairo */
+        #define PANGO_SCALE 1024
+        typedef ... PangoLayout;
+        typedef enum {
+            PANGO_ALIGN_LEFT,
+            PANGO_ALIGN_CENTER,
+            PANGO_ALIGN_RIGHT
+        } PangoAlignment;
+        typedef struct PangoRectangle {
+          int x;
+          int y;
+          int width;
+          int height;
+        } PangoRectangle;
+        PangoLayout *pango_cairo_create_layout (cairo_t *cr);
+        void pango_cairo_update_layout (cairo_t *cr, PangoLayout *layout);
+        void pango_cairo_show_layout (cairo_t *cr, PangoLayout *layout);
+        void pango_layout_set_alignment (
+            PangoLayout *layout, PangoAlignment alignment);
+        void pango_layout_set_markup (
+            PangoLayout *layout, const char *text, int length);
+        void pango_layout_get_pixel_extents (PangoLayout *layout,
+            PangoRectangle *ink_rect, PangoRectangle *logical_rect);
+        PangoFontDescription *pango_font_description_new (void);
+        void pango_font_description_free (PangoFontDescription *desc);
+        void pango_font_description_set_family (PangoFontDescription *desc,
+            const char *family);
+        void pango_font_description_set_size (PangoFontDescription *desc,
+            int size);
+        void pango_layout_set_font_description (PangoLayout *layout,
+            const PangoFontDescription *desc);
+    ''')
+    if (platform.system() == 'Windows'):
+      defaultLibs = {
```

Comment:
Yeah, I was asking about the library names. If that's the only way to do it, then I guess we have to live with it.